python/lsst/meas/algorithms/stamps.py

```diff
@@ -291,7 +295,15 @@ def readFitsWithOptions(cls, filename, options):
         options : `PropertyList`
             Collection of metadata parameters
         """
-        raise NotImplementedError
+        # Load metadata to get class
+        metadata = afwFits.readMetadata(filename, hdu=0)
+        type_name = metadata["STAMPCLS"]
```

Comment:
Consider doing:

```python
type_name = metadata.get("STAMPCLS")
if type_name is None:
    raise RuntimeError(f"No class name in file {filename}. Unable to instantiate correct stamps subclass."
                       " Is this an old version format Stamps file?")
```

Without that check you'll get a confusing `KeyError("STAMPCLS")` which won't be helpful to the general user.

One other thing that worries me a little is that since this is no longer an abstract method, it's possible for a subclass to decide not to implement it themselves. If that was to happen things would blow up really spectacularly since I think you'd get a recursion error.

Maybe short circuit that by doing something like:

```python
if cls is not StampsBase:
    raise NotImplementedError(f"Please implement specific FITS reader for class {cls}")
```
as an equivalent to the abstract method that was there before.

Reply:
Thanks. Added in the error message and some tests to look for them.